src/video_player.rs

```diff
@@ -134,7 +146,13 @@ where
 
                 for msg in inner.bus.iter() {
                     match msg.view() {
-                        gst::MessageView::Error(err) => panic!("{:#?}", err),
+                        gst::MessageView::Error(err) => {
+                            if let Some(ref on_error) = self.on_error {
+                                shell.publish(on_error(&err.error()))
+                            } else {
+                                panic!("{:#?}", err)
```

Comment:
Instead of panicking here, I think it would be better to unconditionally log the error, and then publish the error message if applicable. You can add the `log = 0.4` crate as the logging facade for this (in hindsight this should have always been a log).

Reply:
I hope you don't mind if I use [`tracing`](https://docs.rs/tracing/latest/tracing) instead. It is [compatible](https://docs.rs/tracing/latest/tracing/#log-compatibility) with `log` while providing more functionality.

Reply:
From those docs, it looks like you need to enable the `log` feature on `tracing` so that it also emits to `log`. Also, though I appreciate having more functionality, we're really only logging simple errors in a couple places so only using `log` would likely help reduce binary size. Applications using `tracing` can still consume our `log` records via `tracing-log`.

Reply:
Fair enough, switched to log